opal/mca/accelerator/cuda/accelerator_cuda.c

```diff
@@ -103,9 +120,31 @@ opal_accelerator_base_module_t opal_accelerator_cuda_module =
     accelerator_cuda_get_device_pci_attr,
     accelerator_cuda_device_can_access_peer,
 
-    accelerator_cuda_get_buffer_id
+    accelerator_cuda_get_buffer_id,
+
+    accelerator_cuda_wait_stream,
+    accelerator_cuda_get_num_devices,
+    accelerator_cuda_get_mem_bw
 };
 
+static int accelerator_cuda_get_device_id(CUcontext mem_ctx) {
+    /* query the device from the context */
+    int dev_id = -1;
+    CUdevice ptr_dev;
+    cuCtxPushCurrent(mem_ctx);
+    cuCtxGetDevice(&ptr_dev);
+    for (int i = 0; i < opal_accelerator_cuda_num_devices; ++i) {
```

Comment:
Does this logic work as expected if `CUDA_VISIBLE_DEVICES` is set, i.e. only expose device 0,1,3 but not 2?

Reply:
Yes. The devices numbering is consecutive starting at 0 to the number of visible devices, independent of the device numbers in `CUDA_VISIBLE_DEVICES`.